keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -275,6 +275,19 @@ class SecurityGroupHandler(
         val ingressGroup = rule.securityGroup
         val ingressRange = rule.range
         val protocol = rule.protocol!!.clouddriverProtocolToKeel()
+
+        /**
+         * Due to edge cases in AWS, it is possible for an inbound rule to reference a security group
+         * that no longer exists. When this happens, the clouddriver response won't have a [name]
+         * field. We silently ignore these inbound rules, since they have no effect.
+         */
+        ingressGroup?.let { group ->
+          if(group.name == null) {
+            log.warn("security group $name ($accountName, $region) has inbound rule that references non-existent security group ${group.id} (${group.accountName}, ${group.region}, ${group.vpcId})")
+            return@flatMap emptyList()
```

Comment:
Isn't this going to cause us to ignore all other ingress rules as well? I thought the idea was just to skip this particular one (which you could do with a `.filter`)?

Reply:
I agree that `.filter` is clearer, but I didn't go with it here because I wanted to log the warning, and I didn't see a nice way to combine the two.

Note that the`return@flatMap` call does not short-circuit processing of elements in the `inboundRules` list: it returns from one block iteration, but the block still gets called for subsequent elements of the list.

I was a little confused about this behavior as well, but I confirmed it with a simple example. Check out this [Kotlin playground snippet](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS40LjAiLCJjb2RlIjoiLyoqXG4gKiBGaWx0ZXIgb3V0IHRoZSBldmVuIG51bWJlcnNcbiAqL1xuZnVuIG9kZHMobHN0IDogTGlzdDxJbnQ+KSA9IFxuICAgIGxzdC5mbGF0TWFwIHsgbiAtPlxuICAgICAgXG4gICAgICAgIGlmKG4gJSAyID09IDApIHsgXG4gICAgICAgICAgICByZXR1cm5AZmxhdE1hcCBlbXB0eUxpc3QoKVxuICAgICAgICB9IFxuXG4gICAgICAgIGxpc3RPZihuKVxuICAgIH1cbiBcbmZ1biBtYWluKCkge1xuICAgIHByaW50bG4ob2RkcyhsaXN0T2YoMSwyLDMpKSlcbn0iLCJwbGF0Zm9ybSI6ImphdmEiLCJhcmdzIjoiIn0=) I wrote to confirm the behavior.

Reply:
Thanks for checking! I still think this would be easier to read without the return. Something like:
```kotlin
inboundRules = inboundRules.flatMap { rule ->
  val ingressGroup = rule.securityGroup
  // if the ingress rule references another security group, the the referenced SG's name is null, ignore it with a warning
  if (ingressGroup != null && ingressGroup.name == null) {
    log.warn(...)
    null
  } else {
    // ...
  }
}.filterNotNull()

Reply:
I refactored the code to improve readability.